src/intrusive.rs

```diff
@@ -274,11 +337,13 @@ impl<T> Inner<T> {
     }
 }
 
+#[cfg(all(feature = "std", not(feature = "critical-section")))]
```

Comment:
imo it would be a good idea to put `ListLock` into a separate file and mark the whole file with `#[cfg(...)]`, this would make this easier to read.

Reply:
I've inlined the entire construct into `with_inner`.

Reply:
oh that's quite neat, thank you.